docs/guide/vite.md

```diff
@@ -0,0 +1,383 @@
+
+```javascript
+'@apostrophecms/asset': {
+  options: {
+    productionSourceMaps: true  // Exposes source maps in production
+  }
+}
+```
+  <template v-slot:caption>
+    app.js
+  </template>
+</AposCodeBlock>
+
+::: tip
+Source maps slightly increase your build size but are invaluable for debugging production issues. Consider your specific needs when enabling this option.
+:::
+
+## Extending Vite Configuration
+
+ApostropheCMS's Vite integration provides two methods to customize your Vite configuration. You can either configure it through your module's code or via a dedicated configuration file in your project root.
+
+### Method 1: Module Configuration
+
+Use the `build.vite` property in your module to extend Vite's configuration:
+<AposCodeBlock>
+
+```javascript
+module.exports = {
+  build: {
+    vite: {
+      myViteConfig: {
+        // Vite configuration options
+        define: {
+          __API_URL__: JSON.stringify('https://api.example.com'),
+          __ANALYTICS_ENDPOINT__: JSON.stringify(process.env.ANALYTICS_URL)
+        },
+        css: {
+          // CSS configuration options
+          preprocessorOptions: {
+            scss: {
+              additionalData: `$injectedColor: orange;`
+            }
+          }
+        },
+        optimizeDeps: {
+          // Dependencies optimization options
+          include: ['lodash', 'moment']
+        },
+        server: {
+          // Development server options
+          https: true,
+          port: 3000
+        }
+      }
+    }
+  }
+};
+```
+  <template v-slot:caption>
+    modules/analytics/index.js
+  </template>
+</AposCodeBlock>
+
+### Method 2: Project Configuration File
+
+For project-wide Vite settings, create either `apos.vite.config.js` (ESM) or `apos.vite.config.mjs` (CJS) in your project root:
+
+<AposCodeBlock>
+
+```javascript
+import { defineConfig } from '@apostrophecms/vite/vite';
+
+export default defineConfig({
+  // All standard Vite configuration options are supported
+  plugins: [
+    // Add Vite plugins
+    svgLoader({
+      svgoConfig: {/* custom SVG optimization options */}
+    })
+  ],
+  resolve: {
+    alias: {
+      '@components': '/src/components',
+      '@styles': '/src/styles'
+    }
+  },
+  build: {
+    rollupOptions: {
+      // Customize underlying Rollup bundle
+      output: {
+        manualChunks: {
+          vendor: ['react', 'react-dom']
+        }
+      }
+    }
+  }
+});
+```
+  <template v-slot:caption>
+    apos.vite.config.js
+  </template>
+
+</AposCodeBlock>
+
+## Common Configuration Options
+
+Here are some frequently used Vite configuration options:
+
+- **define**: Replace constants in your code during build time
+- **plugins**: Add Vite plugins for additional functionality
+- **resolve.alias**: Create import aliases for cleaner imports
+- **css**: Configure CSS processing and preprocessors
+- **build**: Customize build output and bundling behavior
+- **server**: Configure development server settings
```

Comment:
We recommend to not touch those, they are handled by Apostrophe

Reply:
Added an extra comment that they shouldn't usually be touched.